### PR TITLE
Add support for cuda-device-side print().

### PR DESCRIPTION
**Pull request by stefanseefeld**
This is a new attempt at adding support for CUDA 'print'. Tests now pass even with ENABLE_CUDASIM=1.

### CODE REVIEW

File: numba/cuda/tests/cudapy/test_print.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+
+def printfloat():
+    i = cuda.grid(1)
+    print(i, 23, 34.3, 321)
+
+
+def cuprintary(A):
+    i = cuda.grid(1)
+    print("A[", i, "]", A[i])
```
Comment (gmarkall) [inline numba/cuda/tests/cudapy/test_print.py]: If you remove the strings from this print statement, does `test_print_array` pass (and can it be not skipped?)?

**Comment by gmarkall** [comment]
I think this looks good - just a quick question about the tests on the diff.

File: numba/cuda/tests/cudapy/test_print.py (hunk 1 above)
Comment (stefanseefeld) [inline numba/cuda/tests/cudapy/test_print.py]: On 27/08/15 09:49 AM, Graham Markall wrote:
> > +def cuprintary(A): > + i = cuda.grid(1) > + print("A[", i, "]", A[i])
>
> If you remove the strings from this print statement, does
> |test_print_array| pass (and can it be not skipped?)?
>

Yes it does. On the other hand, the way the test is written right now is
a good reminder for us to add string support, so I think it's useful to
keep it the way it is until we have added that.


-- 

      ...ich hab' noch einen Koffer in Berlin...

Comment (gmarkall) [inline numba/cuda/tests/cudapy/test_print.py]: OK - noted as https://github.com/numba/numba/issues/1392

**Pull request merged**